aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -492,8 +495,70 @@ void addr_kernel_cuda(TensorIterator &iter, Scalar beta, Scalar alpha) {
   });
 }
 
+// This reduction accumulates results as the type `acc_t`. By default, when
+// `scalar_t` is complex, `acc_t` is the downgraded real number type.
+// Otherwise, `acc_t` and `scalar_t` are the same type.
+template <typename scalar_t, typename acc_t=typename scalar_value_type<scalar_t>::type, typename out_t=typename scalar_value_type<scalar_t>::type>
+void linalg_vector_norm_kernel_cuda_impl(TensorIterator& iter, Scalar val) {
+  double ord_val;
+  if (val.isIntegral(false)) {
+     ord_val = val.to<int64_t>();
+  } else if (val.isFloatingPoint()) {
+     ord_val = val.to<double>();
+  } else {
+     AT_ERROR("linalg_vector_norm_kernel_cuda_impl expects norm to be integer or float");
+  }
+  if (iter.numel() != 0) {
+    if (ord_val == static_cast<double>(0)) {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, NormZeroOps<scalar_t, acc_t>(), 0);
+    } else if (ord_val == static_cast<double>(1)) {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, NormOneOps<scalar_t, acc_t>(), 0);
+    } else if (ord_val == static_cast<double>(2)) {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, NormTwoOps<scalar_t, acc_t>(), 0);
+    } else if (ord_val == static_cast<double>(INFINITY)) {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, AbsMaxOps<scalar_t, acc_t>(), 0);
+    } else if (ord_val == static_cast<double>(-INFINITY)) {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, AbsMinOps<scalar_t, acc_t>(), std::numeric_limits<acc_t>::infinity());
+    } else {
+      gpu_reduce_kernel<scalar_t, out_t>(iter, NormOps<scalar_t, acc_t>{ static_cast<acc_t>(ord_val) }, 0);
```

Comment:
What happens if a large integer value is given that's not representable as a double?

Reply:
That's an interesting point. If ord is a large enough integer value, converting it to a double will introduce some error:
```
>>> ord = torch.tensor(2**62 - 10, dtype=torch.long)
>>> ord.to(torch.float64).to(torch.long) - ord
tensor(10)
```

However, with an ord value this large, I think the norm result will almost always be either infinity (if abs value of any input element is slightly greater than 1), 1 (if abs value of at least one element is exactly 1, and all are less than or equal to 1), or 0 (if abs value of all elements is less than 1). At the moment, I don't see any other possible result.

```
>>> ord = torch.tensor(2**62 - 10, dtype=torch.long)
>>> torch.linalg.norm(torch.tensor([1, 0.1, 0, 1.00000001], dtype=torch.double), ord)
tensor(inf, dtype=torch.float64)
>>> torch.linalg.norm(torch.tensor([1, 0.1, 0, 0], dtype=torch.double), ord)
tensor(1., dtype=torch.float64)
>>> torch.linalg.norm(torch.tensor([0.23, 0.1, 0.999999999999999, 0], dtype=torch.double), ord)
tensor(0., dtype=torch.float64)
```

So I'm inclined to suggest that the error introduced when using a very large integer ord value probably won't have any practical consequences. What do you think?